test/subshell/test_runner_test.go

```diff
@@ -34,16 +34,25 @@ func TestMockingRunner(t *testing.T) {
 
 	t.Run("MockCommitMessage", func(t *testing.T) {
 		t.Parallel()
+		dir := t.TempDir()
 		runner := subshell.TestRunner{
-			WorkingDir: t.TempDir(),
-			HomeDir:    t.TempDir(),
-			BinDir:     filepath.Join(t.TempDir(), "bin"),
+			WorkingDir: dir,
+			HomeDir:    dir,
+			BinDir:     filepath.Join(dir, "bin"),
 		}
 
 		runner.MockCommitMessage("test commit message")
 
-		res := runner.MustQuery("bash", "-c", "$GIT_EDITOR output; cat output")
-		must.Eq(t, "test commit message", res)
+		// Simulate Git calling the mock editor configured by MockCommitMessage and
+		// verify its effect.
+		// MockCommitMessage sets the GIT_EDITOR environment variable to force Git to
```

Comment:
Technically `MockCommitMessage` doesn't set this environment variable. It sets the `gitEditor` property on its `TestRunner` instance. `TestRunner.QueryWithCode` then sets the environment variable in the subshell it calls into, if the property is set.